Add support for `Astro.currentLocale` (#1841)
withastro/starlight: docs/src/content/docs/guides/i18n.mdx

```

Learn more about content collection schemas in [“Defining a collection schema”](https://docs.astro.build/en/guides/content-collections/#defining-a-collection-schema) in the Astro docs.

## Accessing the current locale

> You can read the current locale from the `Astro` global that is available in all contexts in `.astro` files with the [`currentLocale`](https://docs.astro.build/en/reference/api-reference/#astrocurrentlocale) property.
> 
> The following example reads the current locale and uses it to generate a link to an About page in the current language:

Reviewer: ```suggestion
You can use [`Astro.currentLocale`](https://docs.astro.build/en/reference/api-reference/#astrocurrentlocale) to read the current locale in `.astro` components.

The following example reads the current locale and uses it to generate a link to an about page in the current language:
```
